conformance/utils/kubernetes/portset.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// PortSet is a wrapper for handling sets of ports.
+type PortSet []v1alpha2.PortNumber
+
+func (s *PortSet) AssignAvailablePort() (v1alpha2.PortNumber, bool) {
+	if len(*s) == 0 {
+		return 0, false
+	}
+
+	newPort := v1alpha2.PortNumber((*s)[0])
+	*s = (*s)[1:]
+	return newPort, true
+}
```

Comment:
Although more heavyweight, it may be good to just use the [apimachinery sets util](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/sets/int.go) or recreate some subset of that here for the PortNumber type. A `PopAny()` function name like that uses would probably be more familiar than `AssignAvailablePort()`.